### PR TITLE
Don't unmute logs if an error happened

### LINKED ISSUE #923: Compile errors hard to find in output
skaffold dev outputs logs from Pods in the cluster so when updating a service and getting compile errors the errors flow up in the terminal and can be hard to find.

🚀 Moonshot: It would be really cool if you could enter commands into skaffold while it was running to turn off Pod logging dynamically

### PR DESCRIPTION
**Pull request by dgageot**
I'm not fully happy with the code because it introduces some duplication
But that could be dealt with in another PR

Fix #923 

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/runner/runner.go [hunk 1]
```diff
@@ -207,23 +207,41 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*v1
 	changed := changes{}
 	onChange := func() error {
 		logger.Mute()
+		unmute := true
```
Comment (balopat) [inline pkg/skaffold/runner/runner.go]: I would rename this to `hasError`, I think that would be more readable, what do you think?

Comment (balopat) [reply pkg/skaffold/runner/runner.go]: ping @dgageot ?

Comment (dgageot) [reply pkg/skaffold/runner/runner.go]: Yes, `hasError` is much better! I though I pushed this fix but I must have lost it.

**Comment by dgageot** [comment]
Rebasing and making sure the kokoro tests pass

**Pull request merged**